src/services/users.js

```diff
@@ -182,5 +185,118 @@ class UsersService {
       res.boom.badImplementation();
     }
   }
+
+  async addProducts (req, res) {
+    try {
+      const user = await User.findById(req.session.passport.user.id).exec();
+      if (!user) {
+        return res.boom.notFound('User not found');
+      }
+
+      const reqProduct = await this.#productService.findById(
+        req.body.productId
+      );
+      if (!reqProduct) {
+        return res.boom.badRequest('Product does not exist');
+      }
+      if (
+        !reqProduct.sizes.some(
+          (sizeId) => String(sizeId) === String(req.body.sizeId)
+        )
+      ) {
+        return res.boom.badRequest('Product not available in this size');
+      }
+      const userCart = user.cart;
+      const productInCart = userCart.find(
+        (element) =>
+          String(element.productId) === String(req.body.productId) &&
+          String(element.sizeId) === String(req.body.sizeId)
+      );
+      if (productInCart) {
+        await User.updateOne(
+          {
+            _id: req.session.passport.user.id,
+            'cart.productId': req.body.productId,
+            'cart.sizeId': req.body.sizeId
+          },
+          {
+            $set: {
+              'cart.$.quantity': productInCart.quantity + req.body.quantity
+            }
+          }
+        ).exec();
+      } else {
+        await User.updateOne(
+          { _id: req.session.passport.user.id },
+          { $push: { cart: req.body } }
+        ).exec();
+      }
+      return res.json(true);
+    } catch (err) {
+      return res.boom.badImplementation(err);
+    }
+  }
+
+  async #findProductsByQuery (reqQuery, user) {
+    const cart = user.cart;
+
+    const query = {};
```

Comment:
Object.keys(reqQuery).forEach(key => !reqQuery[key] && delete reqQuery[key]) instead of all these ifs